src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -772,7 +772,28 @@ private void rollCycleTo(final int cycle, boolean suppressEOF) {
 
         if (!suppressEOF) {
             assert queue.writeLock().locked();
-            store.writeEOF(wire, timeoutMS());
+
+            if (store.writeEOF(wire, timeoutMS())) {
+                Wire w = this.wire;
+
+                for (int eofCyle = cycle - 1; eofCyle >= queue.firstCycle(); eofCyle--) {
```

Comment:
I believe we already have it somewhere (normalizeEOFs) and it would be great to figure out why it did not work.

Reply:
Sure - you can run the test case and see. The test case is in ChronicleQueueEnterpise. see https://github.com/ChronicleEnterprise/Chronicle-Queue-Enterprise/pull/563

Reply:
I'd prefer we fixed the root cause than adding this complexity, also this will make rolling very expensive, probably negating any gains had by enabling `earlyAcquireNextCycle`.